### PR TITLE
Frontend Translations

### LINKED ISSUE #64: Feat: Translation 2B: Implement Frontend Static Localization & Language Switching
This ticket implements **frontend-side static localization** for UI text, labels, and non-dynamic copy. It complements the backend dynamic translation system by ensuring the user interface itself supports Thai and English consistently.

The frontend will consume translated API responses while managing its own static text translations locally.

This will be for both Web and Mobile

---

### Localization Framework Setup

- Integrate a frontend localization solution (e.g. i18n library)
- Define supported languages:
  - English (default)
  - Thai
- Configure language fallback behavior

---

### Static Text Translation

- Translate all static UI text including:
  - Buttons
  - Labels
  - Navigation items
  - Empty states
  - Error messages
- Organize translations in a maintainable structure (e.g. per feature or screen)

---

### Language Selection & Persistence

- Implement a way for users to select their preferred language
- Persist language preference:
  - Local storage
  - User profile (if applicable)
- Ensure language selection updates UI immediately

---

### Backend Integration

- Ensure frontend sends language preference to backend consistently
- Validate backend responses are rendered without additional client-side translation
- Avoid duplicating backend-translated strings in frontend translation files

---

### UX & Accessibility Considerations

- Ensure:
  - No layout breakage for Thai text
  - Proper font support
  - Consistent formatting across languages
- Validate RTL/LTR assumptions (even if only LTR languages are currently supported)

---

## Testing & Validation

- Verify:
  - Language switching works across all screens
  - Static and dynamic text remain in sync
  - No untranslated strings remain in the UI
- Cross-browser and device testing

---

## Deliverables

- Frontend localization framework configured
- Static translation files for English and Thai
- Language switcher implemented
- Persisted language preference
- Verified integration with backend translation system
- Documentation for adding future languages

### PR DESCRIPTION
**Pull request by AlenGanopolsky**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/64)
Added i18n library to frontend alongside corresponding json blobs for english and thai. Language switching functionality was added to the language settings page - when a user switches their language, static text is converted from english to thai and vice versa - those translations are stored in said json blobs. Language persistence was added using AsyncStorage, but the implementation will use AuthContext once that is merged in. Lastly, language header parameterization was added to backend requests - the backend no longer exclusively uses en-US as the Accept-Language header

# How Has This Been Tested?
Tested frontend on Expo (IOS) and Android Studio

## Screenshots
Please provide screenshots of manual testing (Scalar, frontend pages, etc.)
dashboard_android
event_card_android_thai
maps_android_thai
profile_android_thai
saved_events_android_thai
billing_android_thai
contacts_android_thai
add_contact_android_thai
settings_android_thai
language_settings_android_thai
dashboard_ios
event_card_ios
saved_events_ios
profile_ios
settings_ios
language_settings_ios
add_contact_ios
contact_ios

# Checklist

## General
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [ ] I have written unit tests for my code
- [ ] I have written integration tests where appropriate
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [ ] I have tested error handling and edge cases
- [ ] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]

### Android
[Add Android screenshots here]

### CODE REVIEW

**Review by adescoteaux1 — COMMENTED** [review]
this is so cool!

File: frontend/packages/api-client/src/apiClient.ts [hunk 1]
```diff
@@ -3,11 +3,20 @@ import axios, {
   AxiosResponse,
   InternalAxiosRequestConfig,
 } from 'axios';
+export let currentLanguage = 'en';
+export function setCurrentLanguage(lang: string) {
+  // DEBUG
+  console.log('[apiClient] setCurrentLanguage called', { prev: currentLanguage, next: lang });
```
Comment (adescoteaux1) [inline frontend/packages/api-client/src/apiClient.ts]: remove

File: frontend/packages/api-client/src/apiClient.ts [hunk 2]
```diff
@@ -34,8 +43,10 @@ const getBaseURL = () => {
       return process.env.NEXT_PUBLIC_API_BASE_URL;
     }
   }
+
+  console.warn("using default")
   
-  return 'http://localhost:8080';
+  return 'http://10.110.10.192:8080';
```
Comment (adescoteaux1) [inline frontend/packages/api-client/src/apiClient.ts]: switch back

**Review by adescoteaux1 — COMMENTED** [review]
remove debugs! overwise looks good i think

File: frontend/apps/mobile/app/_layout.tsx [hunk 3]
```diff
@@ -16,28 +18,37 @@ import {
...
-
 export default function RootLayout() {
   const colorScheme = useColorScheme();
+  const [langReady, setLangReady] = useState(false);
   const [loaded, error] = useFonts({
     NunitoSans_400Regular,
     NunitoSans_500Medium,
     NunitoSans_600SemiBold,
     NunitoSans_700Bold,
+    ...MaterialIcons.font,
```
Comment (adescoteaux1) [inline frontend/apps/mobile/app/_layout.tsx]: what are we using this for?

**Review by josh-torre — COMMENTED** [review]
See @adescoteaux1 's changes

**Review by josh-torre — APPROVED** [review]
LGTM 🚀
